src/Controls/src/Core/Page/Page.cs

```diff
@@ -449,6 +449,26 @@ protected virtual void LayoutChildren(double x, double y, double width, double h
 		/// </summary>
 		protected virtual void OnAppearing()
 		{
+			UpdateMenuBarItemBindingContext();
+		}
+
+		/// <summary>
+		/// Resets the BindingContext for each MenuBarItem when it becomes null after page navigation.
+		/// </summary>
+		private void UpdateMenuBarItemBindingContext()
+		{
+			if (MenuBarItems is null)
+			{
+				return;
+			}
+
+			foreach (var menuBarItem in MenuBarItems)
+			{
+				if (menuBarItem.BindingContext is null)
+				{
+					SetInheritedBindingContext(menuBarItem, BindingContext);
```

Comment:
Isn't the BindingContext already propagated to MenuBarItems here?
https://github.com/dotnet/maui/blob/d541b54292554f270919e1c491fae7618a58ea62/src/Controls/src/Core/Page/Page.cs#L496

Reply:
@jsuarezruiz, the BindingContext of MenuBarItem is set to null after navigating to another page. As a result, when returning to the page, the MenuFlyoutItem loses its binding, causing the Command to stop functioning properly. Therefore, the BindingContext of MenuBarItem is updated within the OnAppearing() method of the page.

Reply:
How is it set the first time when you navigate to the page?